docs/future.md

```diff
@@ -11,6 +11,8 @@ alpha(3); // false
 check(alpha, 3); // false
 ```
 
+Any object which contains a `.test()` method (like regular expressions) can be used as an rtype.
```

Comment:
Might make more sense to explain why we were able to use a regex in the example immediately prior to the example instead of farther down in the doc.